### ISSUE TITLE
推送相同app streamid rtmp流到 不同vhost 导致 srs crash

### ISSUE THREAD

**Issue by wnpllrzodiac** [ossrs/srs#983]
当推送流1时，正常。再推送流2至srs时，流2推送失败，同时流1也中断。
服务端**发现 segment fault**

1 rtmp://192.168.1.1:1935/slive/test
2 rtmp://192.168.1.1:1935/slive?vhost=test.abc.com/test

部分srs配置如下：
```
vhost test.abc.com {
    http_remux {
        enabled         on;
        mount           [vhost]/[app]/[stream].flv;
        hstrs           on;
    }
}
vhost __defaultVhost__ {
    http_remux {
        enabled         on;
        mount           [vhost]/[app]/[stream].flv;
        hstrs           on;
    }
}
```

=================部分日志如下=======================
```
[2017-09-14 11:31:13.877][trace][107642][257] RTMP client ip=192.168.6.165
[2017-09-14 11:31:13.878][trace][107642][257] simple handshake success.
[2017-09-14 11:31:13.878][trace][107642][257] connect app, tcUrl=rtmp://192.168.1.1/slive?vhost=test.abc.com, pageUrl=, swfUrl=, schema=rtmp, vhost=test.abc.com, port=1935, app=slive, args=null
[2017-09-14 11:31:13.936][trace][107642][257] client identified, type=fmle-publish, stream_name=tysxlive_t2, duration=-1.00
[2017-09-14 11:31:13.956][trace][107642][257] source url=test.abc.com/slive/test, ip=192.168.6.165, cache=1, is_edge=0, source_id=-1[-1]
[2017-09-14 11:31:13.983][error][107642][257][11] http: multiple registrations for /slive/test.flv. ret=4001(Resource temporarily unavailable)
[2017-09-14 11:31:13.983][error][107642][257][11] http: mount flv stream for vhost=test.abc.com/slive/test failed. ret=4001(Resource temporarily unavailable)
[2017-09-14 11:31:13.983][error][107642][257][11] handle on publish failed. ret=4001(Resource temporarily unavailable)
[2017-09-14 11:31:13.983][error][107642][257][11] notify publish failed. ret=4001(Resource temporarily unavailable)
[2017-09-14 11:31:13.983][trace][107642][257] cleanup when unpublish
```

**此处 segment fault**

==================
src/app/srs_app_http_stream.cpp L871 附近
```
void SrsHttpStreamServer::http_unmount(SrsSource* s, SrsRequest* r)
{
    std::string sid = r->get_stream_url();

    if (sflvs.find(sid) == sflvs.end()) {
        srs_info("ignore unmount flv stream for disabled");
        return;
    }

    SrsLiveEntry* entry = sflvs[sid];
    entry->stream->entry->enabled = false; # ###  // crash here
}
```

**Comment by winlinvip** [comment]
Generally speaking, the default vhost should be deleted when there are other specific vhosts. However, it should not crash and will be fixed in version 3.0, as this is not a typical use case.

`TRANS_BY_GPT3`

**Comment by winlinvip** [comment]
Fixed in SRS3.
